curve/curve.go

```diff
@@ -520,101 +521,81 @@ func (sc StarkCurve) SignFelt(msgHash, privKey *felt.Felt) (*felt.Felt, *felt.Fe
 	return xFelt, yFelt, nil
 }
 
-// HashElements calculates the hash of a list of elements using the StarkCurve struct and a golang Pedersen Hash.
-// (ref: https://github.com/seanjameshan/starknet.js/blob/main/src/utils/ellipticCurve.ts)
-//
+// HashElements calculates the hash of a list of elements using a golang Pedersen Hash.
 // Parameters:
 // - elems: slice of big.Int pointers to be hashed
 // Returns:
 // - hash: The hash of the list of elements
-// - err: An error if any
-func (sc StarkCurve) HashElements(elems []*big.Int) (hash *big.Int, err error) {
+func HashElements(elems []*big.Int) (hash *big.Int) {
+	feltArr := utils.BigIntArrToFeltArr(elems)
 	if len(elems) == 0 {
-		elems = append(elems, big.NewInt(0))
+		feltArr = append(feltArr, new(felt.Felt))
```

Comment:
Do we not have to set the felt here, new(felt.Felt).SetUint(0)?

Reply:
Not really. The default value of felt.Felt already is zero ([0,0,0,0])

Simple test:
![image](https://github.com/user-attachments/assets/cbcef481-c647-44a6-917b-949498c912b0)